File: src/frontend/src/types/knowledge.ts
/** 知识库可见性 */
export type KnowledgeVisibility = 'private' | 'public';

/** 知识库中的文件 */
export interface KnowledgeFile {
  id: string;
  filename: string;
  size: number;
  mime_type: string;
  uploaded_at: string;
}

/** 知识库 */
export interface KnowledgeBase {
  id: string;
  name: string;
  description?: string;
  visibility: KnowledgeVisibility;
  files: KnowledgeFile[];
  file_count: number;
  created_at: string;
  updated_at: string;
}

/** 群组中可用的知识库（含 owner 信息） */
export interface GroupKnowledgeBase {
  id: string;
  name: string;
  description?: string;
  visibility: KnowledgeVisibility;
  username: string;
  file_count: number;
  created_at: string;
  updated_at: string;
}
export interface CreateKnowledgeBaseRequest {
  name: string;
  description?: string;
  visibility?: KnowledgeVisibility;
}

/** 更新知识库请求 */
export interface UpdateKnowledgeBaseRequest {
  name?: string;
  description?: string;
  visibility?: KnowledgeVisibility;
}

File: src/backend/internal/model/knowledge.go
package model

import "time"

// KnowledgeBase 知识库模型
type KnowledgeBase struct {
	ID          string          `json:"id" db:"id"`
	UserID      string          `json:"user_id" db:"user_id"`
	Name        string          `json:"name" db:"name"`
	Description string          `json:"description" db:"description"`
	Visibility  string          `json:"visibility" db:"visibility"`
	CreatedAt   time.Time       `json:"created_at" db:"created_at"`
	UpdatedAt   time.Time       `json:"updated_at" db:"updated_at"`
	Files       []KnowledgeFile `json:"files" db:"-"`
	FileCount   int             `json:"file_count" db:"file_count"`
	Username    string          `json:"username,omitempty" db:"username"`
}

// KnowledgeFile 知识库文件模型
type KnowledgeFile struct {
	ID              string    `json:"id" db:"id"`
	KnowledgeBaseID string    `json:"knowledge_base_id" db:"knowledge_base_id"`
	Filename        string    `json:"filename" db:"filename"`
	FilePath        string    `json:"-" db:"file_path"`
	FileSize        int64     `json:"size" db:"file_size"`
	MimeType        string    `json:"mime_type" db:"mime_type"`
	CreatedAt       time.Time `json:"uploaded_at" db:"created_at"`
}
